### ISSUE TITLE
v8 Historical-Fit-Only Hierarchical Action-Value Candidate — Future Accepted-Bet PnL

### ISSUE THREAD

**Issue by phead198708** [phead198708/BiGan#171]
## Goal

Build and freeze a new hierarchical action-value candidate using only pre-freeze historical training lineage, then evaluate it on a newly collected, chronologically later, market-disjoint window by cost-aware net PnL from bets actually accepted by the unchanged v8 execution layer.

This issue follows the rejected #169 candidate. The #169 future market identities, decisions, labels, settlements, PnL, family/side attribution, and supplemental diagnostics are locked as holdout/report-only evidence and are forbidden from fitting, feature selection, model selection, calibration, threshold selection, guard changes, or sizing changes.

## Candidate lineage

Candidate name: historical_fit_only_hierarchical_action_value_v2

Predeclared action grid:

- BUY_UP_HOLD_TO_SETTLEMENT
- BUY_DOWN_HOLD_TO_SETTLEMENT
- BUY_UP_SELL_BEFORE_CLOSE
- BUY_DOWN_SELL_BEFORE_CLOSE
- NO_TRADE

Predeclared model structure:

1. one deterministic regularized HTS expected-net-return head shared across UP/DOWN HTS rows
2. one deterministic regularized SBC expected-net-return head shared across UP/DOWN SBC rows
3. side/action identity remains an input; no side or family is globally disabled
4. NO_TRADE has deterministic value zero
5. final action ranking compares calibrated cost-aware expected net return from the two family heads plus NO_TRADE
6. existing execution threshold, execution guard, sizing, cost model, and exposure rules remain unchanged

No post-hoc rule derived from #169 family, side, action, or market outcomes may be added.

## Historical split protocol

Before training, write a hash-frozen exclusion registry containing:

- every #169 future market ID and corpus hash
- every #169 decision/shadow/settlement/evaluation artifact hash
- every later future holdout run already designated report-only
- forbidden paths and run IDs

Use only historical direct-training corpus that predates the earliest excluded future decision.

Chronological, market-grouped split:

1. historical_train
2. historical_calibration
3. historical_validation

Hard requirements:

- market IDs disjoint across all splits
- timestamps strictly chronological
- model fitting only on historical_train
- calibration only on historical_calibration
- historical_validation is one-time report-only eligibility evidence
- no validation or future labels used to select features, parameters, thresholds, guard behavior, or sizing
- full decision-time provenance with max_input_ts <= decision_ts
- recursive forbidden-field audit for inference inputs
- outcome/PnL allowed only as post-settlement training targets in historical train/calibration/validation rows

## Target and costs

Primary target: execution_realizable_cost_aware_net_return_per_notional

It must include:

- executable entry price
- executable exit/settlement value according to action family
- spread
- fee
- volatility slippage
- liquidity impact
- queue/fill realizability for SBC
- zero value for NO_TRADE

SBC rows that do not have an executable decision-time-defined exit path must remain fail-closed/ineligible; theoretical exits may not be treated as realized executable return.

## Fixed model protocol

Use one predeclared shallow deterministic XGBoost regressor per trade family with fixed, hash-recorded hyperparameters. No search over #169, historical validation, or future evidence.

Required lineage hashes:

- exclusion registry
- source corpus manifest
- action grid
- feature schema
- target schema
- split manifest
- train/calibration/validation dataset
- fixed model protocol
- HTS model
- SBC model
- calibration artifact
- unchanged execution configuration
- final freeze manifest

## Historical eligibility gates

All gates are predeclared before historical validation is opened:

- zero split overlap
- zero excluded-future-market overlap
- zero feature timestamp violations
- zero forbidden inference-field violations
- finite/bounded model outputs
- minimum train/calibration/validation support by family and side
- positive historical-validation selected accepted-bet net PnL after unchanged guard
- positive historical-validation ROI
- candidate historical-validation net PnL greater than frozen raw/constant baseline
- calibration/high-score support gates pass
- action-family support/profitability gates pass without disabling a family post hoc
- market-level bootstrap and leave-one-market-out diagnostics reported
- largest-winner removal reported
- candidate remains fail-closed if any gate fails

Historical eligibility is necessary only to freeze a diagnostic candidate; it is never promotion evidence.

## Future unseen evaluation

Only after the candidate/config/execution hashes are frozen:

1. collect a new chronologically later read-only public-provider corpus
2. require market disjointness from every training/calibration/validation/#169 market
3. build decision inputs outcome-blind
4. run frozen candidate and frozen baseline
5. apply unchanged threshold, execution guard, sizing, costs, and exposure state
6. record accepted shadow bets
7. load official settlement only after shadow closes
8. reconcile once by hash-pinned identities
9. never refit or alter the candidate from this future result

Predeclared minimum discussion support:

- at least 30 future unique markets
- at least 30 accepted candidate bets
- at least 10 accepted bets per UP/DOWN side
- all accepted bets settled
- zero provenance violations
- zero missing required runtime fields
- zero forbidden outcome-field usage before reconciliation
- candidate cost-aware accepted-bet net PnL > 0
- candidate ROI > 0
- candidate net PnL > frozen baseline
- chronological max drawdown reported
- market bootstrap interval reported
- leave-one-market-out and largest-winner-removal diagnostics reported

These gates may not be lowered after future outcomes are seen.

## Required reports

- historical_action_value_exclusion_registry.json
- hierarchical_action_value_split_manifest.json
- hierarchical_action_value_training_report.json/md
- hierarchical_action_value_calibration_report.json/md
- hierarchical_action_value_historical_validation_report.json/md
- hierarchical_action_value_leakage_audit.json/md
- hierarchical_action_value_freeze_manifest.json
- future collection/input/shadow/settlement/reconciliation manifests
- hierarchical_action_value_future_accepted_bet_pnl_report.json/md
- frozen supplemental diagnostics that cannot mutate the primary gate

## Safety

Always keep false:

- source_model_candidate_eligible
- freeze_ready for production
- promotion_evidence_eligible
- v8_execution_handoff_allowed
- #134_resume_allowed
- #146_start_allowed
- capital_at_risk
- polymarket_write_enabled
- wallet_signing_enabled

paper_only=true. No live trading, wallet signing, Polymarket writes, capital at risk, threshold relaxation, guard relaxation, or score mutation.

## Execution plan

1. Audit current historical corpus and create the immutable exclusion/split protocol.
2. Build causal five-action historical rows and family-specific cost-aware targets.
3. Train fixed HTS/SBC heads and train-only/calibration-only calibration.
4. Run untouched historical validation and hard gates.
5. If and only if historical gates pass, freeze the candidate and all hashes.
6. Collect a new future/disjoint window.
7. Run outcome-blind shadow and exactly-once post-shadow settlement reconciliation.
8. Compare actual accepted-bet cost-aware net PnL against the frozen baseline.
9. Reject or retain the candidate strictly from the predeclared future gates; never tune from the result.

## Validation

- PYTHONPATH=src:. python -m pytest tests/v8 -q
- PYTHONPATH=src python -m ruff check src/bigan/v8/polymarket examples/v8 tests/v8
- git diff --check
- git diff --cached --check

**Comment by phead198708** [comment]
#171 implementation and first real historical gate run are complete. The predeclared candidate is rejected before future collection; no validation-driven retuning was performed.

Implemented:
- immutable #169 future-evidence exclusion registry
- hash verification for the outcome-blind future market registry and nine key #169 artifacts
- chronological market-grouped train/calibration/validation split
- deterministic HTS and SBC XGBoost heads
- NO_TRADE fixed value zero
- calibration-split-only family affine calibration
- untouched validation accepted-bet replay through the unchanged v8 execution threshold/guard/sizing
- raw probability baseline comparison
- market bootstrap, leave-one-market-out, largest-winner removal, leakage, safety, and freeze reports
- outcome-blind inference fails before prediction unless every historical gate passes

Real run:
- run_id: `v8-historical-hierarchical-action-value-20260715T012408Z`
- source: 65 historical markets, 237 decisions, 1,185 five-action rows
- train: 39 markets / 142 decisions
- calibration: 13 markets / 46 decisions
- validation: 13 markets / 49 decisions
- split market overlap: 0
- chronology validation: passed
- #169 excluded future markets: 30
- historical/#169 overlap: 0
- historical max decision timestamp strictly before #169 minimum: true
- feature timestamp violations: 0
- forbidden decision-input violations: 0
- #169 outcome/PnL values loaded for fitting/tuning: false

Untouched historical validation:
- candidate accepted bets: 2
- accepted markets: 2
- accepted sides: UP 1 / DOWN 1
- accepted family: HTS 2 / SBC 0
- cost basis: 0.300000
- candidate net PnL: **-0.302100**
- candidate ROI: **-1.007000**
- baseline accepted bets/net PnL: 0 / 0.000000
- candidate minus baseline: **-0.302100**
- market bootstrap 95% interval: [-0.322100, -0.282100]
- selected actions before guard: NO_TRADE 38, BUY_DOWN_HTS 6, BUY_UP_HTS 5
- model signals above the unchanged 0.02 threshold: 3
- guard-accepted: 2

Calibration:
- validation raw MAE: 0.284324
- validation calibrated MAE: 0.309658
- calibration worsened both calibration-split SBC error and untouched validation aggregate error

Hard-gate blockers:
- `insufficient_historical_validation_accepted_bet_support`
- `insufficient_historical_validation_side_support`
- `insufficient_historical_validation_family_support`
- `historical_validation_candidate_net_pnl_not_positive`
- `historical_validation_candidate_roi_not_positive`
- `historical_validation_candidate_not_better_than_baseline`
- `historical_validation_calibration_mae_worse_than_raw`

Decision:
- `historical_validation_gate_passed=false`
- `candidate_frozen_for_future_evaluation=false`
- `future_collection_allowed=false`
- no new future holdout will be consumed by this candidate
- no threshold, guard, sizing, feature, model, or calibration parameter will be changed from this validation result

Artifact SHA-256:
- exclusion registry: `40c968cd976f7385e19ccd5946cce67780ed4dbdd1031ff4b9785c9862c015b2`
- split manifest: `53a6a6c447b6d341de47ce2393cfd6ac5106174d32c615b6b56117cc3fbca79f`
- HTS model: `6fcea221f60c71d4bc891347dc316a33c957d3bc5f2eb975a3e85db6e81ad3a0`
- SBC model: `4fc872bd5501da37681e030d809f88fd0854b32b337567e8e82b8b18f8a34dad`
- calibration artifact: `365414f2d8d7163f205a5e838d1cf48dc0697116b268d63aa097cfffef8e90dd`
- validation report: `ce62f546d5fe0d1e9b8df26271a6c48815b2dbf871b3360dfcdf802c0acf4322`
- leakage audit: `29cd24512c26da795d91d83128f2717a40af83e83b82d9bb83a9aa5a041f2e9b`
- freeze manifest: `d166e3ead4f4234e709aab0cd61458427f3477895965539953831a0a49eec831`

Validation:
- new focused tests: passed
- full `tests/v8`: passed
- full v8 Ruff: passed
- git diff check: passed

Safety remains blocked: source/freeze/promotion/handoff/#134/#146 false, paper_only=true, capital_at_risk=false, writes/signing disabled.

**Comment by phead198708** [comment]
Implementation commit pushed:

- branch: `codex/v8-issue-171-historical-hierarchical-action-value`
- commit: `e98b9c5e7360c253f077a97e6a475f105f78ab29`

The candidate failed its predeclared historical validation gate, so the correct terminal state for this hypothesis is rejected before future collection. No validation result was fed back into model, calibration, threshold, guard, or sizing.

#171 is complete as a fail-closed candidate experiment; follow-up work must begin in a new planned issue with a genuinely independent training-data/model hypothesis.

**Comment by phead198708** [comment]
Correction: the authoritative pushed commit SHA is `e98b9c56fbbd6536182f451769b1228b2432884c`.
